src/vellum/workflows/nodes/experimental/tool_calling_node/utils.py

```diff
@@ -128,13 +135,39 @@ def execute_function(self) -> BaseNode.Outputs:
         outputs = json.loads(outputs)
         arguments = outputs["arguments"]
 
-        # Call the original function directly with the arguments
-        result = function(**arguments)
+        # Call the function based on its type
+        if is_workflow_class(function):
+            # Dynamically define an Inputs subclass of BaseInputs
+            Inputs = type(
+                "Inputs",
+                (BaseInputs,),
+                {"__annotations__": {k: type(v) for k, v in arguments.items()}},
+            )
```

Comment:
[blocking] we should be able to infer the Inputs class from the workflow class `function`